packages/api/cms-api/src/page-tree/blocks/dam-file-download-link.block.ts

```diff
@@ -0,0 +1,109 @@
+import {
+    AnnotationBlockMeta,
+    BlockContext,
+    BlockData,
+    BlockField,
+    BlockInput,
+    BlockMetaField,
+    BlockMetaFieldKind,
+    createBlock,
+    inputToData,
+    TraversableTransformResponse,
+} from "@comet/blocks-api";
+import { IsEnum, IsOptional, IsUUID } from "class-validator";
+import { FilesService } from "src/dam/files/files.service";
+
+export enum OpenFileTypeMethod {
+    NEW_TAB = "NEW_TAB",
+    DOWNLOAD = "DOWNLOAD",
+}
+
+class DamFileDownloadLinkBlockData extends BlockData {
+    fileId?: string;
+
+    @BlockField({ type: "enum", enum: OpenFileTypeMethod })
+    openFileType: OpenFileTypeMethod;
+
+    async transformToPlain(
+        { filesService }: { filesService: FilesService },
+        { previewDamUrls }: BlockContext,
+    ): Promise<TraversableTransformResponse> {
+        const ret: TraversableTransformResponse = {};
+
+        ret.openFileType = this.openFileType;
+
+        if (this.fileId === undefined) {
+            return ret;
+        }
+
+        const file = await filesService.findOneById(this.fileId);
+
+        if (file) {
+            ret.file = {
+                id: file.id,
+                name: file.name,
+                damPath: await filesService.getDamPath(file),
```

Comment:
In PixelImageBlock etc. we load the DAM path lazy in the Admin for performance reasons. We should also do this here.